File: app/src/main/java/org/openstaxcollege/android/adapters/ShelfRecyclerViewAdapter.java
/**
 * Copyright (c) 2017 Rice University
 *
 * This software is subject to the provisions of the GNU Lesser General
 * Public License Version 2.1 (LGPL).  See LICENSE.txt for details.
 */

package org.openstaxcollege.android.adapters;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import org.openstaxcollege.android.R;
import org.openstaxcollege.android.activity.WebViewActivity;
import org.openstaxcollege.android.beans.Content;
import org.openstaxcollege.android.providers.ShelfBooks;
import org.openstaxcollege.android.utils.OSCUtil;

import java.util.ArrayList;

import co.paulburke.android.itemtouchhelperdemo.helper.ItemTouchHelperAdapter;

/** Adapter to properly display bookshelf in RecyclerView
 * @author Ed Woodward
 * */

public class ShelfRecyclerViewAdapter extends RecyclerView.Adapter<ShelfRecyclerViewAdapter.ViewHolder> implements ItemTouchHelperAdapter
{
    private ArrayList<Content> contentList;
    private Context context;

    private int rowLayout;

    public ShelfRecyclerViewAdapter(ArrayList<Content> content, int rowLayout, Context context)
    {
        contentList = content;
        //Log.d("SRVA","Content size = " + contentList.size());
        this.rowLayout = rowLayout;
        this.context = context;
    }

    @Override
    public ShelfRecyclerViewAdapter.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int i)
    {
        View v = LayoutInflater.from(viewGroup.getContext()).inflate(rowLayout, viewGroup, false);
        return new ShelfRecyclerViewAdapter.ViewHolder(v,contentList);
    }

    @Override
    public void onBindViewHolder(ShelfRecyclerViewAdapter.ViewHolder viewHolder, int i)
    {
        //Log.d("SRVA","BVH: Content size = " + contentList.size());
        Content content = contentList.get(i);
        viewHolder.bookTitle.setText(content.getTitle());
        //viewHolder.other.setText(content.getContentString());
        if (viewHolder.logo != null && content.getIcon() != null)
        {
            viewHolder.logo.setImageResource(OSCUtil.getCoverId(content.getIcon(), context));

        }

    }

    @Override
    public int getItemCount()
    {
        return contentList == null ? 0 : contentList.size();
    }

    @Override
    public void onItemDismiss(int position)
    {
        //Log.d("SRVA","Dismiss: Content size = " + contentList.size());
        //Log.d("SRVA","Dismiss: position = " + position);
        if(contentList.size() >= position + 1)
        {
            Content currentContent = contentList.get(position);
            context.getContentResolver().delete(ShelfBooks.CONTENT_URI, "_id="+ currentContent.getId(), null);

            contentList.remove(position);

            notifyItemRemoved(position);
            Toast.makeText(context, currentContent.getTitle() + context.getString(R.string.removed_from_bookshelf), Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public boolean onItemMove(int fromPosition, int toPosition)
    {
        return true;
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener
    {
        private ImageView logo;
        private TextView bookTitle;
        public View view;
        ArrayList<Content> contentList;

        private ViewHolder(View itemView, ArrayList<Content> contentList)
        {
            super(itemView);
            view = itemView;
            this.contentList = contentList;

            logo = (ImageView) itemView.findViewById(R.id.logoView);
            bookTitle = (TextView)itemView.findViewById(R.id.title);
            itemView.setOnClickListener(this);
        }


        @Override
        public void onClick(View v)
        {
            Content content = contentList.get(getAdapterPosition());
            //Log.d("BRVA", "title: " + content.getBookTitle());
            Content bookTitle = OSCUtil.getTitle(content.getBookTitle(), v.getContext());
            if(bookTitle != null)
            {
                content.setBookUrl(bookTitle.getBookUrl());
            }
            if(OSCUtil.isConnected(context))
            {
                Context context = v.getContext();
                Intent wv = new Intent(v.getContext(), WebViewActivity.class);
                wv.putExtra(v.getContext().getString(R.string.webcontent), content);

                context.startActivity(wv);
            }
            else
            {
                Toast.makeText(context, context.getString(R.string.no_connextion), Toast.LENGTH_SHORT).show();
            }
        }


    }
}

File: app/src/main/java/org/openstaxcollege/android/adapters/ShelfRecyclerViewAdapter.kt
/**
 * Copyright (c) 2017 Rice University
 *
 * This software is subject to the provisions of the GNU Lesser General
 * Public License Version 2.1 (LGPL).  See LICENSE.txt for details.
 */

package org.openstaxcollege.android.adapters

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast

import org.openstaxcollege.android.R
import org.openstaxcollege.android.activity.WebViewActivity
import org.openstaxcollege.android.beans.Content
import org.openstaxcollege.android.providers.ShelfBooks
import org.openstaxcollege.android.utils.OSCUtil

import java.util.ArrayList

import co.paulburke.android.itemtouchhelperdemo.helper.ItemTouchHelperAdapter

/** Adapter to properly display bookshelf in RecyclerView
 * @author Ed Woodward
 */

class ShelfRecyclerViewAdapter(private val contentList: ArrayList<Content>?, private val rowLayout: Int, private val context: Context) //Log.d("SRVA","Content size = " + contentList.size());
    : RecyclerView.Adapter<ShelfRecyclerViewAdapter.ViewHolder>(), ItemTouchHelperAdapter
{

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder
    {
        val v = LayoutInflater.from(viewGroup.context).inflate(rowLayout, viewGroup, false)
        return ViewHolder(v, contentList!!)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int)
    {
        //Log.d("SRVA","BVH: Content size = " + contentList.size());
        val content = contentList!![i]
        viewHolder.bookTitle.text = content.title
        //viewHolder.other.setText(content.getContentString());
        if(viewHolder.logo != null && content.icon != null)
        {
            viewHolder.logo.setImageResource(OSCUtil.getCoverId(content.icon!!, context))

        }

    }

    override fun getItemCount(): Int
    {
        return contentList?.size ?: 0
    }

    override fun onItemDismiss(position: Int)
    {
        //Log.d("SRVA","Dismiss: Content size = " + contentList.size());
        //Log.d("SRVA","Dismiss: position = " + position);
        if(contentList!!.size >= position + 1)
        {
            val currentContent = contentList[position]
            context.contentResolver.delete(ShelfBooks.CONTENT_URI, "_id=" + currentContent.id, null)

            contentList.removeAt(position)

            notifyItemRemoved(position)
            Toast.makeText(context, currentContent.title!! + context.getString(R.string.removed_from_bookshelf), Toast.LENGTH_SHORT).show()
        }
    }

    override fun onItemMove(fromPosition: Int, toPosition: Int): Boolean
    {
        return true
    }

    inner class ViewHolder constructor(var view: View, internal var contentList: ArrayList<Content>) : RecyclerView.ViewHolder(view), View.OnClickListener
    {
        val logo: ImageView
        val bookTitle: TextView

        init
        {

            logo = view.findViewById<View>(R.id.logoView) as ImageView
            bookTitle = view.findViewById<View>(R.id.title) as TextView
            view.setOnClickListener(this)
        }


        override fun onClick(v: View)
        {
            val content = contentList[adapterPosition]
            //Log.d("BRVA", "title: " + content.getBookTitle());
            val bookTitle = OSCUtil.getTitle(content.bookTitle, v.context)
            if(bookTitle != null)
            {
                content.bookUrl = bookTitle.bookUrl
            }
            if(OSCUtil.isConnected(context))
            {
                val context = v.context
                val wv = Intent(v.context, WebViewActivity::class.java)
                wv.putExtra(v.context.getString(R.string.webcontent), content)

                context.startActivity(wv)
            }
            else
            {
                Toast.makeText(context, context.getString(R.string.no_connextion), Toast.LENGTH_SHORT).show()
            }
        }


    }
}
